src/ape/utils/misc.py

```diff
@@ -319,13 +320,68 @@ def inner(*args, **kwargs):
     return inner
 
 
+T = TypeVar("T")
+
+
+def _mergesort_iterators(
+    a: Iterator[T],
+    b: Iterator[T],
+    key: Callable[[T, T], bool] = operator.ge,  # type: ignore
```

Comment:
Option: we could avoid this `type: ignore` and another one below by importing the constraint:

```python
if TYPE_CHECKING:
    from _operator import _SupportsComparison
```

and then adjust the definition of `T` to be bound by it:

```python
T = TypeVar("T", bound="_SupportsComparison")
```

Reply:
Oh yes! I was trying to figure out how to do that!